### PR TITLE
Add Contributor ladder, contributor role, governance and roadmap document

### PR DESCRIPTION
**Pull request by kashifest**
This PR adds the following documents:

- Contributor ladder
- Contributor Role
- Governance 
- Roadmap

### CODE REVIEW

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:34]:
```suggestion
    Cluster Api Provider Metal3's [CONTRIBUTING.md]](https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/CONTRIBUTING.md)
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:38]:
```suggestion
  Anybody that is participating in the Metal3 community is welcome. Any
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:48]: CFP should not be shortened here.

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:47]: PR should not be shortened here.

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:53]: This desperately needs Oxford comma.

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:69]:
```suggestion
[additional roles](CONTRIBUTOR-ROLES.md) within the project and can work more
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:70]:
```suggestion
independently, for example CI tests are automatically triggered in their Pull Requests.
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:94]: Anyone can leave reviews, how is this a privilege?

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:95]:
```suggestion
   * Can recommend other contributors to become Metal3 GitHub Org members
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:96]:
```suggestion
   * Can nominate themselves to [other roles](CONTRIBUTOR-ROLES.md) within Metal3
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:101]:
```suggestion
1. Assign the issue to a [community maintainer](OWNERS).
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:105]:
```suggestion
A Reviewer has responsibility for specific code, documentation,
```
No other role had `Description:`

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:119]: Remove this empty line.

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:127]:
```suggestion
   test failures, or through contributions
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:128]: Can we make it 10 merged PRs? It simpler as a single PR can include numerous commits

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:134]:
```suggestion
   * Can `/lgtm` Pull Requests in specific directories
```
It is not a GH right per se, maybe just omit the type.

We also make it either Prow-specific (tools dependant, specific), or remove the backticks and just make it LGTM (tool independent choice)

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:140]:
```suggestion
 persons GitHub handle under `reviewers:` section.
```

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:157]:
```suggestion
   * Engage/drive
```
It is already stated that the responsibilities include Reviewer responsibilities

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:161]:
```suggestion
   * Same as Reviewers
```
Though I might even to remove it. A higher level always has the same as previous level anyways.

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:168]: This is copy paste from Reviewers, which was already included.

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:172]:
```suggestion
   * Can `/approve` and `/lgtm` Pull Requests in specific
```
Same as in reviewer case

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:185]: This doesn't make clear who opens the PR. The only logical person is the approver recommending a new one.

File: CONTRIBUTOR-LADDER.md
Comment (tuminoid) [inline CONTRIBUTOR-LADDER.md:191]:
```suggestion
GitHub handle under `emeritus_approvers:` or `emeritus_reviewers:` section
```

**Review by tuminoid — COMMENTED** [review]
Comments on the ladder file. More to come for other files a bit later.

File: CONTRIBUTOR-LADDER.md
Comment (smoshiur1237) [reply CONTRIBUTOR-LADDER.md:185]: It should be better option if an approver opens the PR to add new approver or reviewer.

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:16]: This is still unclear if talking about the team or the lead, especially when some of the responsibilities are team's and some is lead's.

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:21]: This is lead's task, to avoid any mixed communication.

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:41]:
```suggestion
1. Nominated by the Security Lead and seconded by an Metal3 Org Admin
```
I think any member along with any maintainer is not right.

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:47]: Not a fan of merging "does not exist", so let's get agreement that if we need such list or not, and then create the group and adjust the PR?

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:62]: We should make it clear what privileges you actually get. This isn't very helpful right now. CI access gives you really broad rights, as you need GH side rights, Jenkins rights, Prow rights?

If we are not ready for open CI team, then maybe drop it from the document until we are?

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:73]:
```suggestion
   * Making sure the repositories are ready to be released
```

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:74]:
```suggestion
   * Performing minor and patch releases
```

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:80]:
```suggestion
   * A proper understanding of release process of different Metal3 repositories
```

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:84]:
```suggestion
   * Release team requires elevated access to Metal3 GitHub organization through
```

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:85]:
```suggestion
   the `metal3_release_team` GitHub team.
```

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:89]: Again, very loose definition. Org Admin would make more sense IMO, since that is required to add the person to the team as well.

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [inline CONTRIBUTOR-ROLE.md:96]: Instead of pointing to that readme, it should point to CONTRIBUTOR-LADDER.md

File: GOVERNANCE.md
Comment (tuminoid) [inline GOVERNANCE.md:9]:
```suggestion
* [Contribution Guide](CONTRIBUTING.md)
```
Why all caps?

**Review by tuminoid — COMMENTED** [review]
Comments on the roles file.

File: ROADMAP.md
Comment (tuminoid) [inline ROADMAP.md:9]: Weirdly the Metal3 roadmap is CAPM3 or BMO... Either we need to re-work the projects per the recently discussed "release layers", or maybe even more preferably the project as a whole, and then per repo/layer separately.

File: ROADMAP.md
Comment (tuminoid) [inline ROADMAP.md:23]:
```suggestion
by opening an issue in the (metal3-docs)[https://github.com/metal3-io/metal3-docs] repository, describing the feature and
```
Let's link top-level repo, or should we link the design to be more precise?

File: ROADMAP.md
Comment (tuminoid) [inline ROADMAP.md:37]:
```suggestion
it is then placed in the `Backlog` column. An issue from the `Backlog` can be
```

File: ROADMAP.md
Comment (tuminoid) [inline ROADMAP.md:10]:
```suggestion
a `Feature Requests` column that contains items that have not yet been
```

File: ROADMAP.md
Comment (tuminoid) [inline ROADMAP.md:33]:
```suggestion
A new feature proposal is moved from the `Feature Requests` column to a
```

File: ROADMAP.md
Comment (tuminoid) [inline ROADMAP.md:42]:
```suggestion
the `Backlog` column, and issues in the `Feature Requests` column that are not
```

**Review by tuminoid — COMMENTED** [review]
Comments on the roadmap.

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:41]: in that case we need to add another role Metal3 Org Admin. I wanted to avoid fine grained classification for now.

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:47]: Yes, the idea is to get some consensus here before merging. I will put it a hold so that it doenst merge. Opinion?

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:89]: check my previous comment.

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:96]: actually I wanted to point to README.

File: ROADMAP.md
Comment (kashifest) [reply ROADMAP.md:9]: Yes those are outdated

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:63]: I want to keep it because its a large part of work. We should think how to make it open towards any org member.

Comment (tuminoid) [reply CONTRIBUTOR-ROLE.md:63]: OK. Let's then make it explicit that it is work in progress, and not just mailing list missing?

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [reply CONTRIBUTOR-ROLE.md:97]: OK, in that case can we more refer to the README? If we create second copy of the same info, but a bit different. Having it in multiple places and especially having content differ just creates confusion.

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [reply CONTRIBUTOR-ROLE.md]: Good point. We have this role right now, so we need to document it as well.

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:63]: I am not intending to make this PR hang on this. We can keep updating the process later on.

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:97]: Yes I can agree to that. I thought of referring the section here but then moved on to put a concise version here to make it consistent with other roles of this document. Do you want me to remove everything I wrote here and just refer to the README?

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md]: Would it be ok that I add an issue here and we discuss the criteria to be selected for that role within maintainers and add it later?

Comment (tuminoid) [reply CONTRIBUTOR-ROLE.md]: That role is not very relevant for a regular contributor, we can add it later. Will you incorporate the original comment anyways?

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [reply CONTRIBUTOR-ROLE.md:63]: No need to hang, agreed. Was just suggesting adding explicit note that opening the CI role is work in progress, so its clear.

File: CONTRIBUTOR-ROLE.md
Comment (tuminoid) [reply CONTRIBUTOR-ROLE.md:97]: IMO not remove everything, leave the responsibilities and privileges part so it is consistent with other roles, but replace the "becoming approver" part with a link to README, since README is all about becoming approver.

**Comment by kashifest** [comment]
@tuminoid your comments are addressed.

**Comment by tuminoid** [comment]
Can we have other review this too?

/cc @dtantsur @elfosardo @lentzi90 @Rozzii

File: CONTRIBUTOR-ROLE.md
Comment (elfosardo) [reply CONTRIBUTOR-ROLE.md:48]: I agree with Tuomo on this, we should create the group first

File: CONTRIBUTOR-ROLE.md
Comment (elfosardo) [inline CONTRIBUTOR-ROLE.md:72]: same here

**Review by elfosardo — CHANGES_REQUESTED** [review]

File: CONTRIBUTOR-ROLE.md
Comment (kashifest) [reply CONTRIBUTOR-ROLE.md:72]: Cool, since we agree, We can create those and I will update PR afterwards

**Review by tuminoid — APPROVED** [review]
/approve

Good job with the documentation!

**Review by Rozzii — APPROVED** [review]
/lgtm

**Pull request merged**
